allennlp/training/deepspeed/checkpointer.py

```diff
@@ -0,0 +1,131 @@
+from typing import Union, Dict, Any, List, Tuple, Optional
+
+import logging
+import os
+import re
+import shutil
+import time
+
+from pathlib import Path
+
+import torch
+
+import allennlp
+from allennlp.nn import util as nn_util
+from allennlp.training import util as training_util, Checkpointer
+
+logger = logging.getLogger(__name__)
+_DeepspeedTrainer = "allennlp.training.deepspeed.trainer.DeepspeedTrainer"
+
+
+class DeepspeedCheckpointer(Checkpointer):
```

Comment:
Does this differ much from the regular checkpointer?

Reply:
Yes, it delegates as much as possible to deepspeed's checkpointer, which is fairly complicated for the reasons mentioned above (mostly model-parallel).